### PR TITLE
Misc. PADle changes 2

### LINKED ISSUE #1742: Write a set of instructions about padle. It should auto send for someone's first-ever padle game, and be available at ^padle help
Feel free to modify the following:

Hello! It looks like this is your first PADle game.  PADle is a wordle-inspired guessing game, but with some differences!
:db: You are not guessing words! Instead, you are guessing monsters!
:db: The difference is based on awakenings, rarity, monster points, and a few other attributes! Your current score screen will tell you if your guesses are correct, wrong, or misplaced (for awakenings).
:db: Because this can be much harder than guessing words, you have infinite guesses!
:db: Share your number of guesses with `[p]padle score`!

To see the main menu at any time, type `[p]padle`. To see this specific greeting message, type `[p]padle help`.

Happy puzzling! :tsubaki:

### LINKED ISSUE #1744: make ^padle stats a ClosableEmbed

### PR DESCRIPTION
**Pull request by ethannp**
- changed filter (and cleaned up the check)
- padle help message (will send on first game played) resolves #1742 
- validrules message
- menu2 for stats resolves #1744

### CODE REVIEW

File: padle/padle.py [hunk 1]
```diff
@@ -44,7 +46,7 @@ class PADle(commands.Cog):
     # Used if no monster list is set
     FALLBACK_PADLE_MONSTER = 3260
     # Set hours to preferred timezone offset from UTC for day change calculations
-    tzinfo = timezone(timedelta(hours = -8.0))
+    tzinfo = timezone(timedelta(hours=-8.0))
```
Comment (RheingoldRiver) [inline padle/padle.py:49]: there's an NA timezone in tsutils

File: padle/padle.py [hunk 2]
```diff
@@ -121,14 +126,41 @@ async def padleadmin(self, ctx):
...
-                       "of more well-known monsters). Everyone is trying to guess the same PAD card. "
-                       "With each guess, you are given feedback as to how similar the two cards are, "
-                       "including comparing the awakenings, rarity, typings, attributes, and monster "
-                       "point sell value.\n- A new PADle is available every day. "
-                       "Use `{}padle start` to begin!".format(prefix))
+        await ctx.send(self._get_help_text(ctx))
+
+    def _get_help_text(self, ctx):
+        args = {"db": get_emoji("db"), "p": ctx.prefix, "tsubaki": get_emoji("tsubaki")}
+        help_text = """PADle is a Wordle-inspired guessing game, but with some differences!
```
Comment (RheingoldRiver) [inline padle/padle.py:133]: put these in a separate file called help_texts and import them from there imo, that'll make these easier to read & edit later

File: padle/padle.py [hunk 3]
```diff
@@ -201,11 +233,17 @@ async def start(self, ctx):
                                        f" use `{prefix}padle resend`.",
                            color=discord.Color.teal())
         if await self.can_play_in_guild(ctx):
+            if len(await self.config.user(ctx.author).all_guesses()) == 0:
+                await ctx.send("Hello! It looks like this is your first PADle game.")
+                await ctx.send(self._get_help_text(ctx))
```
Comment (RheingoldRiver) [inline padle/padle.py:238]: you COULD send these help texts as a SimpleText menu2 if you want (this is optional though) - SimpleText is a precreated menu2 in tsutils like ClosableEmbed, for sending menus that are just text strings and nothing else

File: padle/padle.py [hunk 4]
```diff
@@ -739,23 +775,35 @@ async def filter(self, ctx):
         dbcog = await self.get_dbcog()
         mgraph = dbcog.database.graph
```
Comment (RheingoldRiver) [inline padle/padle.py:776]: btw we usually just say `graph` in most code

File: padle/padle.py [hunk 5]
```diff
@@ -739,23 +775,35 @@ async def filter(self, ctx):
...
+        return (  # Monster is in NA
+                monster.name_en is not None and monster.on_na and
+                # No collab monsters
                 monster.series.series_type != 'collab' and
-                ((monster.sell_mp >= 50000 and (monster.superawakening_count > 1 or prevTrans is not None)) or 
-                 "Super Reincarnated" in monster.name_en) and not monster.is_equip and
-                nextTrans is None and monster.level >= 99)
+                # No equips
+                not monster.is_equip and
+                # No monsters with next transforms
```
Comment (RheingoldRiver) [inline padle/padle.py:799]: fyi this is leaving out like, the "final" stage jiraya and alexander (? i think alexander) - probably not a problem though

Comment (ethannp) [reply padle/padle.py:799]: meh i think the disguises will just be annoying to guess but i can manually add alexander if you want?

Comment (RheingoldRiver) [reply padle/padle.py:774]: nope i dont think it matters, just wanted to point it out

File: padle/view/closable_stats_view.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+from discordmenu.embed.view import EmbedView
+from discordmenu.embed.components import EmbedMain, EmbedThumbnail
+from tsutils.query_settings.query_settings import QuerySettings
+from tsutils.tsubaki.links import MonsterImage
+from tsutils.tsubaki.monster_header import MonsterHeader
+if TYPE_CHECKING:
+    from dbcog.models.monster_model import MonsterModel
+
+
+class ClosableStatsViewProps:
```
Comment (RheingoldRiver) [inline padle/view/closable_stats_view.py:13]: maybe call this PersonalStats

File: padle/view/closable_stats_view.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        self.username = username
+        self.played = played
+        self.win_rate = win_rate
+        self.cur_streak = cur_streak
+        self.max_streak = max_streak
+        self.monster = favorite_monster
+        
+
+class ClosableStatsView:
+    VIEW_TYPE = 'PADleStats'
```
Comment (RheingoldRiver) [inline padle/view/closable_stats_view.py:26]: and PadlePersonalStats for the View_Type string

**Comment by RheingoldRiver** [comment]
P.S. can you ctrl alt L everything? there's a bunch of wrong end-of-files (sorry) (thanks ❤️)

File: padle/padle.py [hunk 8]
```diff
@@ -258,27 +273,21 @@ async def stats(self, ctx):
...
-                max_streak = cur_streak
-        all_monsters_guessed = []
-        for key, value in all_guesses.items():
-            all_monsters_guessed.extend(value)
+            max_streak = max(max_streak, cur_streak)
+        flatten = itertools.chain.from_iterable
+        all_monsters_guessed = list(flatten(all_guesses.values()))
         if not all_monsters_guessed:
             all_monsters_guessed.append(self.FALLBACK_PADLE_MONSTER)
         mode = max(set(all_monsters_guessed), key=all_monsters_guessed.count)
```
Comment (Raijinili) [inline padle/padle.py:281]: This `max` is O(nm), where n is the size of the list and m is the number of unique monsters in it. For each unique monster, you're going through the entire list to count it. It's not a big deal here (10k is small), but be aware.

Prefer a `collections.Counter` when you're trying to count up a lot of things at the same time.

**Pull request merged**